python/semantic_kernel/kernel.py

```diff
@@ -239,7 +237,7 @@ async def invoke_prompt(
         Returns:
             FunctionResult | list[FunctionResult] | None: The result of the function(s)
         """
-        if not arguments:
+        if arguments is None:
```

Comment:
It seems wrong that a KernelArguments instance with a non-null settings would evaluate to false as an empty dict would. This could cause people troubles in other cases as well. Would it make more sense to change this behavior by overriding __bool__() so that it only evaluates to false if the dict is empty and settings is None?

Reply:
In Python, empty structures will evaluate to False. For example, an empty string will evaluate to be False; or an empty list will evaluate False; same with dictionary. This is just how Python does things.

Reply:
Agreed!

I have overridden the `__bool__` function and added unit test for it. However, I am keeping the check `if arguments is None:` to be consistent with the other APIs: `invoke(...)` and `invoke_stream(...)`.